EZP-32097: Expose EzSystemInfo in Admin UI and SystemInfoDumpCommand

konradoboza opened the pull request:
JIRA: https://jira.ez.no/browse/EZP-32097

Excerpt from the public ticket:
>After moving system information related components entirely to ez-support-tools there is an easy way to extend tabs by tagging missing support_tools.system_info.collector.system.ez collector. Thanks to that we can expose DXP Platform information (version, naming, stability, etc) both in the UI and the command. That enables us to improve system version detection on the support portal.

Data gathered by the collector is available via `ibexa` option:
```
php bin/console ez-support-tools:dump-info ibexa
```
TODO:
- [x] added EOL and EOM dates to `IbexaSystemInfo.php` 

<img width="877" alt="Zrzut ekranu 2020-10-21 o 16 16 14" src="https://user-images.githubusercontent.com/34310128/96732794-07764100-13b9-11eb-9f96-2d66fa2bbee7.png">

andrerom commented on src/bundle/Resources/config/default_settings.yaml:
```diff
@@ -2,6 +2,10 @@ parameters:
     ezsettings.default.system_info_view: {}
     ezsettings.global.system_info_view:
         pjax_tab:
+            ez:
+                template: '@@ezdesign/system_info/platform.html.twig'
+                match:
+                    SystemInfo\Identifier: 'ez'
```
As this is for master ee should find other names then ez and platform as of today :)

andrerom replied:
Maybe just:

```suggestion
            ibexa:
                template: '@@ezdesign/system_info/ibexa.html.twig'
                match:
                    SystemInfo\Identifier: 'ibexa'
```

lserwatka commented:
You have show dates when support expire.

andrerom commented:
> You have show dates when support expire.

We can't yet show if customer has subscription _(aka if they have Support or not)_, that is later step where we fetch such info from API on updates/support.ibexa.co. 

But you are defiantly right that we should show the dates for EOM and EOL here, on provide a link to read more about it on https://support.ibexa.co/Public/Service-Life

@konradoboza Feel free to propose new params on value object in order to pass this from collector to template, maybe opt for Date objects as type?

@lserwatka BTW is there any term to refer to all versions _(including OSS)_? Or should we just opt to just have `Ibexa` on tab and `Software info` as header name for instance here?

dew326 commented on src/bundle/Resources/views/themes/admin/system_info/platform.html.twig:
```diff
@@ -0,0 +1,67 @@
+<section class="ez-fieldgroup">
+    <h2 class="ez-fieldgroup__name">{{ 'platform'|trans|desc('Platform') }}</h2>
+    <div class="ez-fieldgroup__content">
+        <table class="table ez-table--list ez-table--to-left">
```
Please add also `ez-table`

dew326 commented on src/bundle/Resources/views/themes/admin/system_info/platform.html.twig:
```diff
@@ -0,0 +1,67 @@
+    <div class="ez-fieldgroup__content">
+        <table class="table ez-table--list ez-table--to-left">
+            <tbody>
+            <tr class="ez-table__row">
```
Indentation

dew326 commented on src/bundle/Resources/views/themes/admin/system_info/platform.html.twig:
```diff
@@ -0,0 +1,67 @@
+{% trans_default_domain "systeminfo" %}
+
+{% set icons = asset('bundles/ezplatformadminui/img/ez-icons.svg') %}
```
We are changing the way of adding icons to the twig, please see this: https://github.com/ezsystems/ezplatform-core/pull/21
I know this is `work in progress` but we will merge it as soon as possible so could you adapt to this and use the twig helper? `{{ ez_icon_path('checkmark') }}`

andrerom commented on src/bundle/Resources/config/services.yaml:
```diff
@@ -57,9 +57,8 @@ services:
-        # And it does not look like there is any way to add it from this package, so maybe it needs to be made extensible(?)
-        #tags: [{ name: "support_tools.system_info.collector", identifier: "ez" }]
+        tags:
+            - { name: "support_tools.system_info.collector", identifier: "ez" }
```
```suggestion
            - { name: "support_tools.system_info.collector", identifier: "ibexa" }
```

andrerom commented on src/bundle/Resources/translations/systeminfo.en.xliff:
```diff
@@ -176,6 +221,11 @@
         <target state="new">System Information</target>
         <note>key: systeminfo</note>
       </trans-unit>
+      <trans-unit id="cf859bea232bcc0f7f5755421888dbc8b285b7a4" resname="tab.name.ez">
+        <source>Platform</source>
+        <target state="new">Platform</target>
+        <note>key: tab.name.ez</note>
+      </trans-unit>
```
```suggestion
      <trans-unit id="cf859bea232bcc0f7f5755421888dbc8b285b7a4" resname="tab.name.ibexa">
        <source>Ibexa</source>
        <target state="new">Ibexa</target>
        <note>key: tab.name.ibexa</note>
      </trans-unit>
```

andrerom commented on src/bundle/Resources/translations/systeminfo.en.xliff:
```diff
@@ -56,6 +56,51 @@
+      <trans-unit id="ba5031b35adb26a0ea1c7d2964e970ff2eac672d" resname="platform.name">
+        <source>Name</source>
+        <target state="new">Name</target>
+        <note>key: platform.name</note>
+      </trans-unit>
+      <trans-unit id="29de330ee1afde5fd7b8b52493b688d499586b79" resname="platform.version">
+        <source>Version</source>
+        <target state="new">Version</target>
+        <note>key: platform.version</note>
+      </trans-unit>
+      <trans-unit id="9d5bd3bd589f34e8261031892dc2009a4e38a25f" resname="platform.is_enterprise">
+        <source>Is Enterprise</source>
+        <target state="new">Is Enterprise</target>
+        <note>key: platform.is_enterprise</note>
+      </trans-unit>
+      <trans-unit id="c1090cf68d66f668176c5fcc00d6f0b4470caa8a" resname="platform.is_commerce">
+        <source>Is Commerce</source>
+        <target state="new">Is Commerce</target>
+        <note>key: platform.is_commerce</note>
+      </trans-unit>
+      <trans-unit id="9173c16987b27c1dbb2235ec2d67a35dce07b296" resname="platform.is_eom">
+        <source>Is End of Maintenance</source>
+        <target state="new">Is End of Maintenance</target>
+        <note>key: platform.is_eom</note>
+      </trans-unit>
+      <trans-unit id="c05720448171561b73112dc677234f6cf7b8275e" resname="platform.is_eol">
+        <source>Is End of Life</source>
+        <target state="new">Is End of Life</target>
+        <note>key: platform.is_eol</note>
+      </trans-unit>
+      <trans-unit id="f6ade203331c09ee732438a4f00ba36e1aa848d7" resname="platform.is_trial">
+        <source>Is Trial</source>
+        <target state="new">Is Trial</target>
+        <note>key: platform.is_trial</note>
+      </trans-unit>
+      <trans-unit id="5c9a8c36f76d9991640a1fa67533358828ffaec6" resname="platform.stability">
+        <source>Stability</source>
+        <target state="new">Stability</target>
+        <note>key: platform.stability</note>
+      </trans-unit>
```
```suggestion
      <trans-unit id="3c72abbe626f27d174d80fc434b851cce7bde6d3" resname=" ibexa.product">
        <source>Product</source>
        <target state="new">Product</target>
        <note>key: ibexa.product</note>
      </trans-unit>
      <trans-unit id="ba5031b35adb26a0ea1c7d2964e970ff2eac672d" resname="ibexa.name">
        <source>Name</source>
        <target state="new">Name</target>
        <note>key: ibexa.name</note>
      </trans-unit>
      <trans-unit id="29de330ee1afde5fd7b8b52493b688d499586b79" resname="ibexa.version">
        <source>Version</source>
        <target state="new">Version</target>
        <note>key: ibexa.version</note>
      </trans-unit>
      <trans-unit id="9d5bd3bd589f34e8261031892dc2009a4e38a25f" resname="ibexa.is_enterprise">
        <source>Is DXP</source>
        <target state="new">Is DXP</target>
        <note>key: ibexa.is_enterprise</note>
      </trans-unit>
      <trans-unit id="c1090cf68d66f668176c5fcc00d6f0b4470caa8a" resname="ibexa.is_commerce">
        <source>Has Commerce</source>
        <target state="new">Has Commerce</target>
        <note>key: ibexa.is_commerce</note>
      </trans-unit>
      <trans-unit id="9173c16987b27c1dbb2235ec2d67a35dce07b296" resname="ibexa.is_eom">
        <source>Is End of Maintenance</source>
        <target state="new">Is End of Maintenance</target>
        <note>key: ibexa.is_eom</note>
      </trans-unit>
      <trans-unit id="c05720448171561b73112dc677234f6cf7b8275e" resname="ibexa.is_eol">
        <source>Is End of Life</source>
        <target state="new">Is End of Life</target>
        <note>key: ibexa.is_eol</note>
      </trans-unit>
      <trans-unit id="f6ade203331c09ee732438a4f00ba36e1aa848d7" resname="ibexa.is_trial">
        <source>Is Trial</source>
        <target state="new">Is Trial</target>
        <note>key: ibexa.is_trial</note>
      </trans-unit>
      <trans-unit id="5c9a8c36f76d9991640a1fa67533358828ffaec6" resname="ibexa.stability">
        <source>Stability</source>
        <target state="new">Stability</target>
        <note>key: ibexa.stability</note>
      </trans-unit>
```

andrerom commented on src/bundle/Resources/views/themes/admin/system_info/platform.html.twig:
```diff
@@ -0,0 +1,67 @@
+
+{% set icons = asset('bundles/ezplatformadminui/img/ez-icons.svg') %}
+
+<!-- Tab name: {{ 'tab.name.ez'|trans|desc('Platform') }} -->
```
```suggestion
<!-- Tab name: {{ 'tab.name.ibexa'|trans|desc('Ibexa') }} -->
```

andrerom commented on src/bundle/Resources/views/themes/admin/system_info/platform.html.twig:
```diff
@@ -0,0 +1,67 @@
+<!-- Tab name: {{ 'tab.name.ez'|trans|desc('Platform') }} -->
+
+<section class="ez-fieldgroup">
+    <h2 class="ez-fieldgroup__name">{{ 'platform'|trans|desc('Platform') }}</h2>
```
```suggestion
    <h2 class="ez-fieldgroup__name">{{ 'ibexa.product'|trans|desc('Product') }}</h2>
```

andrerom commented on src/bundle/Resources/views/themes/admin/system_info/platform.html.twig:
```diff
@@ -0,0 +1,67 @@
+                <td class="ez-table__cell">
+                    <svg class="ez-icon ez-icon--small">
+                        <use xlink:href="{{ icons }}#{{ (info.isEnterpise == true) ? 'checkmark' : 'discard' }}"></use>
+                    </svg>
+                </td>
+            </tr>
+            <tr class="ez-table__row">
+                <td class="ez-table__cell">{{ 'platform.is_commerce'|trans|desc('Is Commerce') }}</td>
+                <td class="ez-table__cell">
+                    <svg class="ez-icon ez-icon--small">
+                        <use xlink:href="{{ icons }}#{{ (info.isCommerce == true) ? 'checkmark' : 'discard' }}"></use>
+                    </svg>
+                </td>
+            </tr>
+            <tr class="ez-table__row">
+                <td class="ez-table__cell">{{ 'platform.is_eom'|trans|desc('Is End of Maintenance') }}</td>
+                <td class="ez-table__cell">
+                    <svg class="ez-icon ez-icon--small">
+                        <use xlink:href="{{ icons }}#{{ (info.isEndOfMaintenance == true) ? 'checkmark' : 'discard' }}"></use>
+                    </svg>
+                </td>
+            </tr>
+            <tr class="ez-table__row">
+                <td class="ez-table__cell">{{ 'platform.is_eol'|trans|desc('Is End of Life') }}</td>
+                <td class="ez-table__cell">
+                    <svg class="ez-icon ez-icon--small">
+                        <use xlink:href="{{ icons }}#{{ (info.isEndOfLife == true) ? 'checkmark' : 'discard' }}"></use>
+                    </svg>
+                </td>
+            </tr>
+            <tr class="ez-table__row">
+                <td class="ez-table__cell">{{ 'platform.is_trial'|trans|desc('Is Trial') }}</td>
+                <td class="ez-table__cell">
+                    <svg class="ez-icon ez-icon--small">
+                        <use xlink:href="{{ icons }}#{{ (info.isTrial == true) ? 'checkmark' : 'discard' }}"></use>
+                    </svg>
+                </td>
+            </tr>
+            <tr class="ez-table__row">
+                <td class="ez-table__cell">{{ 'platform.stability'|trans|desc('Stability') }}</td>
```
```suggestion
                <td class="ez-table__cell">{{ 'ibexa.name'|trans|desc('Name') }}</td>
                <td class="ez-table__cell">{{ info.name }}</td>
            </tr>
            <tr class="ez-table__row">
                <td class="ez-table__cell">{{ 'ibexa.version'|trans|desc('Version') }}</td>
                <td class="ez-table__cell">{{ info.release }}</td>
            </tr>
            <tr class="ez-table__row">
                <td class="ez-table__cell">{{ 'ibexa.is_enterprise'|trans|desc('Is DXP') }}</td>
                <td class="ez-table__cell">
                    <svg class="ez-icon ez-icon--small">
                        <use xlink:href="{{ icons }}#{{ (info.isEnterpise == true) ? 'checkmark' : 'discard' }}"></use>
                    </svg>
                </td>
            </tr>
            <tr class="ez-table__row">
                <td class="ez-table__cell">{{ 'ibexa.is_commerce'|trans|desc('Has Commerce') }}</td>
                <td class="ez-table__cell">
                    <svg class="ez-icon ez-icon--small">
                        <use xlink:href="{{ icons }}#{{ (info.isCommerce == true) ? 'checkmark' : 'discard' }}"></use>
                    </svg>
                </td>
            </tr>
            <tr class="ez-table__row">
                <td class="ez-table__cell">{{ 'ibexa.is_eom'|trans|desc('Is End of Maintenance') }}</td>
                <td class="ez-table__cell">
                    <svg class="ez-icon ez-icon--small">
                        <use xlink:href="{{ icons }}#{{ (info.isEndOfMaintenance == true) ? 'checkmark' : 'discard' }}"></use>
                    </svg>
                </td>
            </tr>
            <tr class="ez-table__row">
                <td class="ez-table__cell">{{ 'ibexa.is_eol'|trans|desc('Is End of Life') }}</td>
                <td class="ez-table__cell">
                    <svg class="ez-icon ez-icon--small">
                        <use xlink:href="{{ icons }}#{{ (info.isEndOfLife == true) ? 'checkmark' : 'discard' }}"></use>
                    </svg>
                </td>
            </tr>
            <tr class="ez-table__row">
                <td class="ez-table__cell">{{ 'ibexa.is_trial'|trans|desc('Is Trial') }}</td>
                <td class="ez-table__cell">
                    <svg class="ez-icon ez-icon--small">
                        <use xlink:href="{{ icons }}#{{ (info.isTrial == true) ? 'checkmark' : 'discard' }}"></use>
                    </svg>
                </td>
            </tr>
            <tr class="ez-table__row">
                <td class="ez-table__cell">{{ 'ibexa.stability'|trans|desc('Stability') }}</td>
```

andrerom reviewed (changes requested):
Suggestions for changes to names and labels

@SylvainGuittard Maybe you should also give some input on naming of things here, especially regarding Enterprise vs DXP, maybe that should be left for 3.3

lserwatka commented:
Ca we name this tab like "My Ibexa", "My Software", "My Installation" or something like this?

andrerom commented:
> Ca we name this tab like "My Ibexa", "My Software", "My Installation" or something like this?

Yes indeed, I so far opted to propose just `Ibexa` as I realized this would be about our product right now, and in the future also info on subscription and services from us for instance. So `My Ibexa` fits well also.

lserwatka commented:
I like "My Ibexa". Let's go for it.

konradoboza commented:
Code review remarks applied, did some refactoring, corrected nitpicks, and tweaked tab's design, updated screenshot below. @dew326 feel free to suggest design adjustments if needed.

![Zrzut ekranu 2020-10-24 o 16 34 13](https://user-images.githubusercontent.com/34310128/97084375-c2e7e100-1616-11eb-9f50-7fb664e8cb39.png)

andrerom commented on src/bundle/SystemInfo/Collector/IbexaSystemInfoCollector.php:
```diff
@@ -117,63 +118,87 @@ public function __construct(SystemInfoCollector $composerCollector, $debug = fal
         }
 
-        $ez->stability = $this->getStability();
+        $ibexa->endOfMaintenanceDate = $this->getEOMDate($ibexaRelease);
+        $ibexa->endOfLifeDate = $this->getEOLDate($ibexaRelease);
```
Won't be accurate if Trial/OSS with the current wording of properties, see logic above

andrerom replied:
But if prefixed to clarify the sates here are only valid if you have a valid DXP/Enterprise subscription, then it would be fine.

Note: Could probably use this variable in logic above a few places where there are checks against the dates.

andrerom replied:
Or, we can solve this with some information text in the table, something like
End of Maintenance*                     <date>** _(if expired then RED color, if OSS then orange otherwise)_
End of Life*                                     <date>** _(if expired then RED color, if OSS then RED if EOM and orange otherwise)_

* You can read more about Service life for Ibxea DXP with a business license here.
if OSS
** Does not apply to open source versions, for guaranteed maintenance, security fixes before disclosure and much more, upgrade to Ibexa DXP.
elseif Trial
* Only applicable if upgrading to a business license, trial license is firmly limited to 6 months.

andrerom replied:
Too complex, we can rather drop the dates for OSS and Trial and just a `*` explaining it with link to service life if people want to see what they get on that if they upgrade to EE

dew326 commented on src/bundle/Resources/views/themes/admin/system_info/my_ibexa.html.twig:
```diff
@@ -0,0 +1,67 @@
+                <tr class="ez-table__row">
+                    <td colspan="2">
+                        <hr/>
+                        * you can read more about Service Life <a target="_blank" href="https://support.ibexa.co/Public/Service-Life">here</a>
```
Shouldn't this be translatable?

dew326 commented on src/bundle/Resources/views/themes/admin/system_info/my_ibexa.html.twig:
```diff
@@ -39,28 +39,31 @@
+                        <td colspan="2">
+                            <hr/>
+                            * {{ 'ibexa.read_more'|trans|desc('You can read more about Service Life for Ibexa DXP with a business license') }}
+                            <a target="_blank" href="https://support.ibexa.co/Public/Service-Life">here</a>
```
`here` should be also translatable.

konradoboza commented:
All code review remarks applied, please re-review @andrerom @dew326 @lserwatka @mateuszdebinski.

andrerom reviewed (approved):
+1

We can iterate on this later to provide further links to OSS version for instance to read more on what you are missing out from.

GrabowskiM commented on src/bundle/Resources/views/themes/admin/system_info/my_ibexa.html.twig:
```diff
@@ -0,0 +1,70 @@
+                    <td class="ez-table__cell">{{ 'ibexa.is_dxp'|trans|desc('Is DXP') }}</td>
+                    <td class="ez-table__cell">
+                        <svg class="ez-icon ez-icon--small">
+                            <use xlink:href="{{ ez_icon_path((info.isEnterprise == true) ? 'checkmark' : 'discard') }}"></use>
```
wouldn't `ez_icon_path(info.isEnterprise ? 'checkmark' : 'discard')` be enough?

(if so, there are two more occurences below)

GrabowskiM commented on src/bundle/Resources/views/themes/admin/system_info/my_ibexa.html.twig:
```diff
@@ -0,0 +1,70 @@
+                    <tr class="ez-table__row">
+                        <td colspan="2">
+                            <hr/>
+                            * {{ 'ibexa.read_more'|trans|desc('You can read more about Service Life for Ibexa DXP with a business license') }}
```
You could put whole link inside translation, there are probably some other languages where link wo'nt be last word

konradoboza replied:
Not sure if I get what you meant, so I explicitly mentioned the URL in the link. Having the whole text as a link doesn't look good. If you suggest rephrasing the whole sentence I am open to that.

andrerom replied:
Maybe reference to use `%link%` inside  the string?

konradoboza replied:
As agreed on Slack, we can leave it as-is for now, possible iterations might be done as followups.

micszo reviewed (approved):
QA Approved on v3.2.0 with branch with OSS, EE & EC.

konradoboza commented:
Updated PR according to @SylvainGuittard suggestions passed via Slack, updated tab shown below:

<img width="1041" alt="Zrzut ekranu 2020-11-3 o 08 58 33" src="https://user-images.githubusercontent.com/34310128/97961095-ee0ac700-1db2-11eb-974f-9e860602d9f4.png">

SylvainGuittard commented:
Excellent! Thanks @konradoboza
